addons/website_sale/controllers/main.py

```diff
@@ -576,16 +576,16 @@ def checkout_form_validate(self, mode, all_form_values, data):
         error = dict()
         error_message = []
 
-        if data.get('partner_id'):
+        if mode[1] == 'billing' and data.get('partner_id'):
             partner_su = request.env['res.partner'].sudo().browse(int(data['partner_id'])).exists()
-            name_change = partner_su and 'name' in data and data['name'] != partner_su.name
-            email_change = partner_su and 'email' in data and data['email'] != partner_su.email
+            name_change = partner_su and partner_su.name and 'name' in data and (data['name'] or False) != (partner_su.name or False)
+            email_change = partner_su and partner_su.email and 'email' in data and (data['email'] or False) != (partner_su.email or False)
 
             # Prevent changing the partner name if invoices have been issued.
             if name_change and not partner_su.can_edit_vat():
                 error['name'] = 'error'
                 error_message.append(_(
-                    "Changing your name is not allowed once invoices have been issued for your"
+                    "Changing your name is not allowed once documents have been issued for your"
```

Comment:
This change requires updating the `.pot`. Also, it this warning not only for invoices? (I didn't check)

Reply:
yess, my bad! I will add it :pray:

Reply:
Juliette thinks this is more proper because not only the invoice but also an SO can stop the name change. I think she has a good point about this and only specifying the invoice can be inaccurate :pray:

Reply:
Alright, I was just curious :)